Use the enum class in ErrorType at XMLErrors

AlexsanderDamaceno opened the pull request:
#### e734aaf3a805927c3e9676e2ecdb3aebdf1319b4
<pre>
Use the enum class in ErrorType at XMLErrors
<a href="https://bugs.webkit.org/show_bug.cgi?id=285331">https://bugs.webkit.org/show_bug.cgi?id=285331</a>

Reviewed by Charlie Wolfe.

Use the enum class in ErrorType at XMLErrors

* Source/WebCore/xml/XMLErrors.cpp:
(WebCore::XMLErrors::handleError):
* Source/WebCore/xml/XMLErrors.h:
* Source/WebCore/xml/parser/XMLDocumentParser.cpp:
(WebCore::XMLDocumentParser::pushCurrentNode):
(WebCore::XMLDocumentParser::handleError):
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::XMLDocumentParser::doWrite):
(WebCore::warningHandler):
(WebCore::fatalErrorHandler):
(WebCore::normalErrorHandler):

Canonical link: <a href="https://commits.webkit.org/289106@main">https://commits.webkit.org/289106@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0d68ab67848df5d6581c3cba0e4d7c50ac5dd7f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83265 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2884 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/37555 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/88345 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34278 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/85356 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2967 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10826 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64781 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22533 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86318 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2161 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75679 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45065 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2066 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30210 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/33327 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73190 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/30600 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89718 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/10533 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7573 "Found 1 new test failure: http/wpt/mediarecorder/record-96KHz-sources.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73217 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10758 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/71499 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/72447 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18331 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16652 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15381 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1871 "Built successfully") | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/22/builds/13058 "The change is no longer eligible for processing.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/10488 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15963 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10340 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13810 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12111 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/291044dc3c1e5d244a6c285325d216cb3a5a6120. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/291044dc3c1e5d244a6c285325d216cb3a5a6120. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5a5c659aeee6bee3c047d2d979556682c999dc16. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5a5c659aeee6bee3c047d2d979556682c999dc16. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

charliewolfe commented on Source/WebCore/xml/XMLErrors.cpp:
```diff
@@ -63,13 +63,13 @@ void XMLErrors::handleError(ErrorType type, const char* message, int lineNumber,
-        case warning:
-            appendErrorMessage("warning"_s, position, message);
+        case ErrorType::Warning:
+            appendErrorMessage("ErrorType::Warning"_s, position, message);
```
This patch probably shouldn’t change the error message. We should just keep the “warning” string, especially since you’re keeping "error" as the message below on line 73.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d964364dd9f03b9a85d82bde2674efc8e4d09fe4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d964364dd9f03b9a85d82bde2674efc8e4d09fe4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

nt1m commented on Source/WebCore/xml/XMLErrors.h:
```diff
@@ -43,7 +43,7 @@ class XMLErrors {
     explicit XMLErrors(Document&);
 
-    enum ErrorType { warning, nonFatal, fatal };
+    enum class ErrorType : uint8_t { Warning, NonFatal, Fatal };
```
```suggestion
    enum class Type : uint8_t { Warning, NonFatal, Fatal };
```

You could shorten this to Type. The outer `XMLErrors` namespace already provides context

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2f1ada27e41516b3b4a7e59e5052fced38a3e3b6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2f1ada27e41516b3b4a7e59e5052fced38a3e3b6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e0d68ab67848df5d6581c3cba0e4d7c50ac5dd7f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e0d68ab67848df5d6581c3cba0e4d7c50ac5dd7f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38489

AlexsanderDamaceno committed:
Use the enum class in ErrorType at XMLErrors
https://bugs.webkit.org/show_bug.cgi?id=285331

Reviewed by Charlie Wolfe.

Use the enum class in ErrorType at XMLErrors

* Source/WebCore/xml/XMLErrors.cpp:
(WebCore::XMLErrors::handleError):
* Source/WebCore/xml/XMLErrors.h:
* Source/WebCore/xml/parser/XMLDocumentParser.cpp:
(WebCore::XMLDocumentParser::pushCurrentNode):
(WebCore::XMLDocumentParser::handleError):
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::XMLDocumentParser::doWrite):
(WebCore::warningHandler):
(WebCore::fatalErrorHandler):
(WebCore::normalErrorHandler):

Canonical link: https://commits.webkit.org/289106@main
